master/internal/api_project.go

```diff
@@ -185,6 +190,83 @@ func (a *apiServer) getProjectColumnsByID(
 	if err != nil {
 		return nil, err
 	}
+
+	trialIDs := make([]int, 0, len(hyperparameters)+1)
+	// dummy value so bun doesn't generate invalid sql
```

Comment:
I'd suggest skipping `trialsQuery` execution `if len(trialIDs) == 0` instead. why hit the database when you know the response will be empty.